docs/basics/101-109-rerun.rst

```diff
@@ -0,0 +1,179 @@
+   $ datalad run -m "create a list of podcast titles" "python code/list_titles.py"
+
+However, think about any situation where the command would be longer than this,
+or that is many months past the first execution. It wouldn't be easy to remember
+the command, nor would it be very convenient to copy it from the ``run-record``.
+
+Luckily, a fellow student remembered the DataLad way of re-executing
+a ``run`` command, and he's eager to show it to you.
+
+"In order to re-execute a ``datalad run`` command,
+find the checksum of its commit and use it as an argument for the
+``datalad rerun`` command! That's it!" he says happily.
+
+So you go ahead and find the commit checksum in your history:
+
+.. runrecord:: _examples/DL-101-108-113
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :lines: 1-12
+   :emphasize-lines: 8
+
+   $ git log -2
+
+Take that checksum and paste it after ``datalad rerun``
+(the first 6-8 characters of the checksum would be sufficient,
+here we're using all of them).
+
+.. runrecord:: _examples/DL-101-108-114
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :realcommand: echo "$ datalad rerun $(git rev-parse HEAD~1)" && datalad rerun $(git rev-parse HEAD~1)
+
+Now DataLad has made use out of the ``run-record``, and based on the information in
+it re-executed the original command. Because we
+updated the script, the output ``Podcasts.tsv`` has changed and now contains the podcast
+titles of both subdirectories.
+You've probably already guessed it, but the easiest way to check whether a ``datalad rerun``
+has changed the desired output file is
+to check whether the rerun command appears in the datasets history: If a ``datalad rerun``
+does not add or change any content in the dataset, it will also not be recorded in the history.
+
+.. runrecord:: _examples/DL-101-108-115
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :emphasize-lines: 4
+
+   git log -1
+
+In the datasets history,
+we can see that a new ``datalad run`` was recorded. This action is
+committed by Datalad under the original commit message of the ``run``
+command, and looks just like the previous ``datalad run`` commit apart
+from the execution time.
+
+Note that ``datalad rerun`` works with ``datalad run`` or ``datalad rerun`` commands
+(yes, you can ``rerun`` a ``datalad rerun``),
+but not with any other type of datalad command in your history
+such as a ``datalad save`` on results or outputs after you executed a script.
+Therefore, make it a
+habit to record the execution of scripts by plugging it into ``datalad run``.
+
+This very basic example of a ``datalad run`` is as simple as it can get, but it
+is already
+convenient from a memory-load perspective: Now you don't need to
+remember the commands or scripts involved in creating an output.
+Also, incidentally, we have generated :term:`provenance` information. It is
+now recorded in the history of the dataset how the output ``Podcasts.tsv`` came
+into existence.
```

Comment:
They may not be ready yet to have their head blown off with a demo of `git blame`, but I think it would be cool to show them `git log -- Podcasts.tsv`, and explain that any dataset is capable of answering the question "who touch this file?", and datalad can tell you exactly how it came to be.